src/lib/APS2.cpp

```diff
@@ -1041,28 +1083,31 @@ void APS2::check_clocks_status() {
  		throw APS2_PLL_LOST_LOCK;
  	}
 
- 	//Now check the FPGA clocks
- 	if (statusRegs.userFirmwareVersion >= 0x212) {
-		FILE_LOG(logDEBUG1) << "User status: " << hexn<8> << statusRegs.userStatus;
-	 	bool clocksGood = true;
-	 	for (auto bit : {MMCM_SYS_LOCK_BIT, MMCM_CFG_LOCK_BIT, MIG_C0_LOCK_BIT, MIG_C0_CAL_BIT, MIG_C1_LOCK_BIT, MIG_C1_CAL_BIT}) {
-	 		clocksGood &= (statusRegs.userStatus >> bit) & 0x1;
-	 	}
-	 	if (!clocksGood) {
-	 		throw APS2_MMCM_LOST_LOCK;
-	 	}
+	//On board MMCM PLL's are reported either through user status or the PLL_STATUS_ADDR CSR register
+	uint32_t pll_status;
+	if (legacy_firmware) {
+		if (statusRegs.userFirmwareVersion < 0x212) return; //nothing to read
```

Comment:
I am also tempted to just drop this check since I don't think any firmware versions that old exist in the wild any longer.